tests/modeltests/tst_disassemblyoutput.cpp

```diff
@@ -0,0 +1,107 @@
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <QDebug>
+#include <QObject>
+#include <QTest>
+#include <QStandardPaths>
+#include <QProcess>
+#include <QString>
+#include <QStringList>
+#include <QVector>
+
+#include <models/disassemblyoutput.h>
+#include "data.h"
+
+class TestDisassemblyOutput : public QObject
+{
+    Q_OBJECT
+private slots:
+    void initTestCase()
+    {
+        qRegisterMetaType<Data::Symbol>();
+    }
+
+    void testSymbol_data()
+    {
+        QTest::addColumn<Data::Symbol>("symbol");
+        Data::Symbol symbol = {"__cos_fma",
+                               4294544,
+                               2093,
+                               "vector_static_gcc/vector_static_gcc_v9.1.0",
+                               "/home/milian/projects/kdab/rnd/hotspot/3rdparty/perfparser/tests/auto/perfdata/vector_static_gcc/vector_static_gcc_v9.1.0",
+                               "/home/milian/projects/kdab/rnd/hotspot/3rdparty/perfparser/tests/auto/perfdata/vector_static_gcc/vector_static_gcc_v9.1.0"};
+
+        QTest::newRow("curSymbol") << symbol;
+    }
+
+    void testSymbol()
+    {
+        QFETCH(Data::Symbol, symbol);
+
+        const auto actualBinaryFile = QFINDTESTDATA(symbol.binary);
+        symbol.actualPath = actualBinaryFile;
+
+        QVERIFY(!actualBinaryFile.isEmpty() && QFile::exists(actualBinaryFile));
+        const auto expectedOutputFile = actualBinaryFile + ".expected.txt";
+        const auto actualOutputFile = actualBinaryFile + ".actual.txt";
+
+        DisassemblyOutput disassemblyOutput = DisassemblyOutput::disassemble("objdump","x86_64", symbol);
+        QStringList actualTextList = QStringList();
+        for (int row = 0; row < disassemblyOutput.disassemblyLines.size(); row++) {
+            const auto &disassemblyLine = disassemblyOutput.disassemblyLines.at(row);
+            actualTextList << QString::number(disassemblyLine.addr) << disassemblyLine.disassembly
+                           << QLatin1String("\n");
+        }
+        QString actualText = actualTextList.join(" ");
+        QVector<DisassemblyOutput::DisassemblyLine> disassemblyLines = disassemblyOutput.disassemblyLines;
+        {
+            QFile actual(actualOutputFile);
```

Comment:
open the file earlier above, then write directly to it via a QTextStream in the loop above instead of first building a list and then joining it